code/Controllers/HomeController.cs

```diff
@@ -45,17 +46,42 @@ public HomeController(BagsApi bagsApi)
 				if (product?.Images != null && product.Images.Count() > 0)
 					model.Images = product.Images
 						.OrderBy(image => image.Priority)
-						.SelectAndSwallowReferenceType(image =>new Uri(image.Large))
+						.SelectAndSwallowReferenceType(image => new Uri(image.Large))
 						.Where(image => image != null);
 
 				if (product?.Tags != null && product.Tags.Count() > 0)
 				{
 					var brand = product.Tags.FirstOrDefault(x => x.IsBrand)?.TagName ?? "";
-					var styles = String.Join("/", product.Tags.Where(x => x.IsStyle).Select(x=> x.TagName).ToArray());
+					var styles = String.Join("/", product.Tags.Where(x => x.IsStyle).Select(x => x.TagName).ToArray());
 
 					model.Title = $"{brand} {product.Name} - {styles}";
 				}
 			}
+			else if (!String.IsNullOrEmpty(tagId))
+			{
+				model.Url = $"https://bagcupid.com/app/{path.TrimStart('/')}";
+
+				var tags = tagId.Split('_');
+				var products = await bagsApi.GetProductsByTags(tags);
+
+				if (products?.Count > 0)
+					model.Images = products.Take(5)
+						.Select(product => product.Images
+							.OrderBy(image => image.Priority)
+							.SelectAndSwallowReferenceType(image => new Uri(image.Large))
+							.Where(image => image != null).FirstOrDefault());
+
+
+				var tagModels = await bagsApi.GetTags();
```

Comment:
Performance: Move the call to `bagsApi.GetTags()` up to the top of the method and save the result in a Task (don't use `await`).  Then down here actually do the await.  This will make it so the query for this starts and is running in the background while we are also fetching the `ProductsByTags`.

Style: Also, we can get rid of one of the if statements along the way:

```c#
else if (...)
{
	model.Url = $"https://bagcupid.com/app/{path.TrimStart('/')}";
	var tags = tagId.Split('_');

	var productsTask = bagsApi.GetProductsByTags(tags);
	var tagModelsTask = bagsApi.GetTags();
	
	model.Images = (await productsTask ?? Enumerable.Empty<BagsApi.Product>())
		.Take(5)
		.Select(product => product.Images
			.OrderBy(image => image.Priority)
			.SelectAndSwallowReferenceType(image => new Uri(image.Large))
			.Where(image => image != null)
			.FirstOrDefault())
		.DefaultIfEmpty(model.Images.First());

	var tagModels = await tagModelsTask;
	...
}
```

The idea behind this is to wait until the last possible moment to `await` on something.  The more time you spend doing other work while waiting, the more efficient the code will be.  This is especially true if you have other network operations to do like in this case, where we need to fetch a couple different things.